olly/middleware.py

```diff
@@ -4,15 +4,27 @@
 from profiles.models import UserProfile, BannedUser
 
 
-class CheckBanListMiddleware:
-    def __init__(self, get_response):
-        self.get_response = get_response
+def tenant_middleware(get_response):
+    def middleware(request):
+
+        host = request.get_host()
+        subdomain = host.split('.')[0]
+        domain = host.split('.')[-2]
+
+        if domain == 'nfmstudios' or subdomain == 'nfm':
+            request.tenant = 'stock'
+
+        else:
+            request.tenant = 'stock'
+
+        # all done, the view will receive a request with a tenant attribute
+        return get_response(request)
+
+    return middleware
 
-    def __call__(self, request):
```

Comment:
why is this being removed? django makes a call to this I believe...?

Reply:
I have tested it and it works properly without it